### ISSUE TITLE
workbox-webpack-plugin: crashing program if using Yarn 2 PnP

### ISSUE THREAD

**Issue by pizzafox** [GoogleChrome/workbox#2385]
**Library Affected**:
`workbox-webpack-plugin`

**Browser & Platform**:
Yarn 2 with PnP, although this should affect any package manager using PnP (including Yarn 1)

**Issue or Feature Request Description**:

> Unveiled in September 2018, Plug'n'Play is a new innovative installation strategy for Node. Based on prior works from other languages (for example autoload from PHP), it presents interesting characteristics that build upon the regular commonjs require workflow in an almost completely backward-compatible way.

Currently, requiring the package while using Yarn 2 with PnP will totally fail because of an improperly `require()`d dependency.
This is because the package `webpack` is not being properly listed in the package.json or it *is* listed but is being improperly referenced. Because of this PnP is erroring.

To reproduce just run the following file when using Yarn 2 with `workbox-webpack-plugin` listed in the dependencies.

```js
require('workbox-webpack-plugin');
```

This error is from requiring `workbox-webpack-plugin` in a Next.js project.

```
> Build error occurred
Error: A package is trying to access a peer dependency that should be provided by its direct ancestor but isn't

Required package: webpack (via "webpack/lib/ModuleFilenameHelpers")
Required by: workbox-webpack-plugin@virtual:8d2229208177a81d3e407c2f6e51911d14a13296098a81cf0520502e8413bc22600d127018f3993e1d07c9e3835db36dc3099ec76aaf12a245d51ab27e2c117f#npm:5.0.0 (via /project/.yarn/$$virtual/workbox-webpack-plugin-virtual-30102c9dca/0/cache/workbox-webpack-plugin-npm-5.0.0-d60977d80b-2.zip/node_modules/workbox-webpack-plugin/build/lib/)

Require stack:
- /project/.yarn/$$virtual/workbox-webpack-plugin-virtual-30102c9dca/0/cache/workbox-webpack-plugin-npm-5.0.0-d60977d80b-2.zip/node_modules/workbox-webpack-plugin/build/lib/get-manifest-entries-from-compilation.js
- /project/.yarn/$$virtual/workbox-webpack-plugin-virtual-30102c9dca/0/cache/workbox-webpack-plugin-npm-5.0.0-d60977d80b-2.zip/node_modules/workbox-webpack-plugin/build/generate-sw.js
- /project/.yarn/$$virtual/workbox-webpack-plugin-virtual-30102c9dca/0/cache/workbox-webpack-plugin-npm-5.0.0-d60977d80b-2.zip/node_modules/workbox-webpack-plugin/build/index.js
- /project/next.config.js
- /project/.yarn/$$virtual/next-virtual-81c60fb54b/0/cache/next-npm-9.2.1-9e2a04bc32-2.zip/node_modules/next/dist/next-server/server/config.js    - /project/.yarn/$$virtual/next-virtual-81c60fb54b/0/cache/next-npm-9.2.1-9e2a04bc32-2.zip/node_modules/next/dist/build/index.js
- /project/.yarn/$$virtual/next-virtual-81c60fb54b/0/cache/next-npm-9.2.1-9e2a04bc32-2.zip/node_modules/next/dist/cli/next-build.js
- /project/.yarn/$$virtual/next-virtual-81c60fb54b/0/cache/next-npm-9.2.1-9e2a04bc32-2.zip/node_modules/next/dist/bin/next
    at Object.makeError (/project/.pnp.js:19286:24)
    at resolveToUnqualified (/project/.pnp.js:28168:35)
    at resolveRequest (/project/.pnp.js:28261:27)
    at Object.resolveRequest (/project/.pnp.js:28329:26)
    at Function.module_1.Module._resolveFilename (/project/.pnp.js:27558:34)
    at Function.module_1.Module._load (/project/.pnp.js:27443:40)
    at Module.require (internal/modules/cjs/loader.js:1044:19)
    at require (internal/modules/cjs/helpers.js:77:18)
    at Object.<anonymous> (/project/.yarn/$$virtual/workbox-webpack-plugin-virtual-30102c9dca/0/cache/workbox-webpack-plugin-npm-5.0.0-d60977d80b-2.zip/node_modules/workbox-webpack-plugin/build/lib/get-manifest-entries-from-compilation.js:12:5)
    at Module._compile (internal/modules/cjs/loader.js:1158:30) {
  code: 'MODULE_NOT_FOUND',
  pnpCode: 'MISSING_PEER_DEPENDENCY',
  data: {
    request: 'webpack/lib/ModuleFilenameHelpers',
    issuer: '/project/.yarn/$$virtual/workbox-webpack-plugin-virtual-30102c9dca/0/cache/workbox-webpack-plugin-npm-5.0.0-d60977d80b-2.zip/node_modules/workbox-webpack-plugin/build/lib/',
    issuerLocator: {
      name: 'workbox-webpack-plugin',
      reference: 'virtual:8d2229208177a81d3e407c2f6e51911d14a13296098a81cf0520502e8413bc22600d127018f3993e1d07c9e3835db36dc3099ec76aaf12a245d51ab27e2c117f#npm:5.0.0'
    },
    dependencyName: 'webpack'
  },
  requireStack: [
    '/project/.yarn/$$virtual/workbox-webpack-plugin-virtual-30102c9dca/0/cache/workbox-webpack-plugin-npm-5.0.0-d60977d80b-2.zip/node_modules/workbox-webpack-plugin/build/lib/get-manifest-entries-from-compilation.js',
    '/project/.yarn/$$virtual/workbox-webpack-plugin-virtual-30102c9dca/0/cache/workbox-webpack-plugin-npm-5.0.0-d60977d80b-2.zip/node_modules/workbox-webpack-plugin/build/generate-sw.js',
    '/project/.yarn/$$virtual/workbox-webpack-plugin-virtual-30102c9dca/0/cache/workbox-webpack-plugin-npm-5.0.0-d60977d80b-2.zip/node_modules/workbox-webpack-plugin/build/index.js',
    '/project/next.config.js',
    '/project/.yarn/$$virtual/next-virtual-81c60fb54b/0/cache/next-npm-9.2.1-9e2a04bc32-2.zip/node_modules/next/dist/next-server/server/config.js',
    '/project/.yarn/$$virtual/next-virtual-81c60fb54b/0/cache/next-npm-9.2.1-9e2a04bc32-2.zip/node_modules/next/dist/build/index.js',
    '/project/.yarn/$$virtual/next-virtual-81c60fb54b/0/cache/next-npm-9.2.1-9e2a04bc32-2.zip/node_modules/next/dist/cli/next-build.js',
    '/project/.yarn/$$virtual/next-virtual-81c60fb54b/0/cache/next-npm-9.2.1-9e2a04bc32-2.zip/node_modules/next/dist/bin/next'
  ]
}
```

**Comment by jeffposnick** [comment]
Hello!

I must confess that I only use `npm`, and I'm not familiar with `yarn`'s PnP behavior.

That being said, I did revisit how we are pulling in some of the `webpack` modules we're dependent on inside of `workbox-webpack-plugin`, and I've updated that to use the "official" exports from the `webpack` package, instead of pulling them in via path specifiers: https://github.com/GoogleChrome/workbox/pull/2397

I'm not sure if that will help with this specific issue or not—my understanding is that the best practice for a module like `workbox-webpack-plugin` that offers a plugin for another module like `webpack` is to list the version compatibility via a `peerDependencies` field (which we're doing already), but not to include the `webpack` module explicitly in `dependencies`. (E.g., that's what [`html-webpack-plugin` does](https://github.com/jantimon/html-webpack-plugin/blob/dfb98e70f813562c007249152ba902e53cbd94e1/package.json#L62-L64).)
